src/sentry/incidents/endpoints/serializers/workflow_engine_incident.py

```diff
@@ -52,49 +97,136 @@ def get_incident_status(self, priority: int | None) -> int:
+                "dateCreated": open_period.date_started,
             }
-        ]
+            open_period_activities.append(resolved)
 
-    def get_alert_rule(self, open_period: GroupOpenPeriod) -> AlertRuleSerializerResponse:
-        # TODO: Implement this
-        return {
-            "id": "-1",
-            "name": "Test Alert Rule",
-            "organizationId": "-1",
-            "status": 1,
-            "query": "test",
-            "aggregate": "test",
-            "timeWindow": 1,
-            "resolution": 1,
-            "thresholdPeriod": 1,
-            "triggers": [
-                {
-                    "id": "-1",
-                    "status": 1,
-                    "dateModified": datetime.now(),
-                    "dateCreated": datetime.now(),
-                }
-            ],
-            "dateModified": datetime.now(),
-            "dateCreated": datetime.now(),
-            "createdBy": {},
-            "description": "test",
-            "detectionType": "test",
+        # look up Activity rows for other status changes (warning / critical)
+        status_change_activities = Activity.objects.filter(
+            group=open_period.group, type=ActivityType.ActivityType.SET_PRIORITY
+        )
+
+        activity_status_to_incident_status = {
+            "high": IncidentStatus.CRITICAL,
+            "medium": IncidentStatus.WARNING,
         }
+        for activity in status_change_activities:
+            priority = activity.data.get("priority")
+            previous_priority = None
+            previous_activities = status_change_activities.filter(datetime__lt=activity.datetime)
+
+            for previous_activity in previous_activities:
+                previous_priority_data = previous_activity.data.get("priority")
+                if (
+                    previous_priority_data != priority
+                    and previous_priority_data in activity_status_to_incident_status.keys()
+                ):
+                    previous_priority = activity_status_to_incident_status.get(
+                        previous_priority_data
+                    )
+                    break
+
+            status_change = {
+                "id": "-1",
+                "incidentIdentifier": incident_identifier,
+                "type": IncidentActivityType.STATUS_CHANGE,
+                "value": activity_status_to_incident_status.get(priority),
+                "previousValue": previous_priority,
+                "user": None,
+                "comment": None,
+                "dateCreated": open_period.date_started,
+            }
+            open_period_activities.append(status_change)
+
+        return open_period_activities
+
+    def get_open_periods_to_detectors(
+        self, open_periods: list[GroupOpenPeriod]
+    ) -> dict[GroupOpenPeriod, Detector]:
+        action_group_statuses = ActionGroupStatus.objects.filter(
+            group__in=[open_period.group for open_period in open_periods]
```

Comment:
would we ever be querying for more than one group? If not, It might be worth validating that all the open periods are for the same group

Reply:
hmmm it takes a list of `GroupOpenPeriod` objects (previously incidents) so yes there could potentially be more than one group. In practice today this endpoint is hit when loading the metric alert rule details page to list past incidents which could have triggered on multiple groups.